smartsim/_core/control/previewrenderer.py

```diff
@@ -0,0 +1,123 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2023, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import typing as t
+from enum import Enum
+
+import jinja2
+
+from ..._core.config import CONFIG
+from ..._core.control import Manifest
+from ...error.errors import PreviewFormatError
+from ...log import get_logger
+
+logger = get_logger(__name__)
+
+if t.TYPE_CHECKING:
+    from smartsim import Experiment
+
+
+_OutputFormatString = t.Optional[t.Literal["plain_text", "html"]]
+_VerbosityLevelString = t.Literal["info", "debug", "developer"]
```

Comment:
1. Do we need both `_VerbosityLevelString` and `Verbosity`?
2. If we add `Verbosity.DEFAULT = "info"` does it allow us to avoid hardcoding defaults (see [line 122](https://github.com/CrayLabs/SmartSim/pull/469/files#diff-3c7fd203fcb3ac4ad98cde50342ee0f7fb41da902f2f806b88c6c92e6012a3bcR122))?

Reply:
1. I like _verbositylevelstring for type stuff. I think you suggested the Verbosity class to replace "info", but it is causing conflicts with _Verbositylevelstring
2.  I was getting a mypy conflict when I tried that:  error: Incompatible return value type (got "Literal[Verbosity.INFO]", expected "Literal['info', 'debug', 'developer']") .